src/Raven.Server/Documents/Handlers/Processors/Attachments/RangeHelper.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Raven.Server.Documents.Handlers.Processors.Attachments
+{
+    internal static class RangeHelper
+    {
+        public static (bool SendBody, long Start, long? Length) SetRangeHeaders(HttpContext context, long size)
+        {
+            var responseHeaders = context.Response.GetTypedHeaders();
+            context.Response.Headers.AcceptRanges = "bytes";
+
+            long start = 0;
+            var (isRangeRequest, range) = ParseRange(context, size);
+            if (isRangeRequest)
+            {
+                if (range is null)
+                {
+                    context.Response.StatusCode = (int)HttpStatusCode.RequestedRangeNotSatisfiable;
+                    responseHeaders.ContentRange = new ContentRangeHeaderValue(size);
+                    responseHeaders.ContentLength = 0;
+                    return (false, start, null);
+                }
+
+                context.Response.StatusCode = (int)HttpStatusCode.PartialContent;
+                responseHeaders.ContentRange = ComputeContentRange(range, size, out start, out var length);
+                responseHeaders.ContentLength = length;
+                return (true, start, length);
+            }
+            else
+            {
+                responseHeaders.ContentLength = size;
+                return (true, 0, null);
+            }
+        }
+
+        private static (bool IsRangeRequest, RangeItemHeaderValue Range) ParseRange(HttpContext context, long length)
+        {
+            var rawRangeHeader = context.Request.Headers.Range;
+            if (StringValues.IsNullOrEmpty(rawRangeHeader))
+            {
+                return (false, null);
+            }
+
+            // Perf: Check for a single entry before parsing it
+            if (rawRangeHeader.Count > 1 || (rawRangeHeader[0] ?? string.Empty).Contains(','))
+            {
+                // The spec allows for multiple ranges but we choose not to support them because the client may request
+                // very strange ranges (e.g. each byte separately, overlapping ranges, etc.) that could negatively
+                // impact the server. Ignore the header and serve the response normally.
+                return (false, null);
```

Comment:
That being the case, shouldn't we return an error?

Reply:
If the server does not respond with status code 206 with the Content-Range Header set, it indicates to the client that the requested range could not be fulfilled and the response contains the complete Stream. 
This is again taken from AspNetCore. If they can ignore it, I'm sure we can also ignore it here.